crates/pixi_command_dispatcher/src/source_metadata/mod.rs

```diff
@@ -202,18 +217,26 @@ impl SourceMetadataSpec {
     async fn compute_input_hash(
         command_queue: CommandDispatcher,
         source: &SourceCheckout,
-        manifest_path: PathBuf,
+        project_model: Option<ProjectModelV1>,
         input_globs: Option<BTreeSet<String>>,
     ) -> Result<Option<InputHash>, CommandDispatcherError<SourceMetadataError>> {
         let input_hash = if source.pinned.is_immutable() {
             None
         } else {
-            // Compute the input hash based on the manifest path and the input globs.
-            let mut input_globs = input_globs.unwrap_or_default();
-            input_globs.insert(manifest_path.to_string_lossy().into_owned());
+            // Compute the input hash based on the project model and the input globs.
+            let input_globs = input_globs.unwrap_or_default();
+            let project_model_hash = project_model.as_ref().map(|project_model| {
```

Comment:
Instead of recomputing lets use the value already computer above. Also slightly stronger guarentee rhat its the same.